### PR TITLE
Retry all curl errors

### PR DESCRIPTION
**Pull request by athomas**
* Pass --retry-all-errors to curl.
* Fix error handling for failed downloads.
* Update outdated file header comment.

### CODE REVIEW

**Comment by athomas** [comment]
Apparently github action doesn't have the right curl version on `ubuntu-latest` yet.

**Comment by mit-mit** [comment]
Please update  the changelog and pubspec version too

File: setup.sh [hunk 1]
```diff
@@ -76,12 +76,19 @@ fi
 echo "Downloading ${URL}..."
 
 # Download installation zip.
-curl --connect-timeout 15 --retry 5 "$URL" > "${HOME}/dartsdk.zip"
-unzip -o "${HOME}/dartsdk.zip" -d "${RUNNER_TOOL_CACHE}" > /dev/null
+curl --connect-timeout 15 --retry 5 --retry-all-errors "$URL" \
```
Comment (athomas) [inline setup.sh:79]: We need to feature test for the retry-all-errors here, otherwise we are incompatible with some older image versions.

**Comment by devoncarew** [comment]
Closing as the `@tool_cache` downloadTool() function we're using now has some auto-retry logic.

**Pull request closed without merging**